lib/Model.js

```diff
@@ -60,6 +60,7 @@ Model.compile = function compile (name, schema, options, base) {
   function NewModel (obj) {
     Model.call(this, obj);
     applyVirtuals(this, schema);
+    schema.parseDynamo(this, schema.toDynamo(this));
```

Comment:
The issue is the Object constructor is used in a lot of places and may not be a full object.  The partial objects passed to  `schema.toDynamo` cause the errors.

During a save (or create), the object is already passed to `schema.toDynamo`.  What needs to happen is that Dynamo Object needs to be parsed.  

Try adding `schema.parseDynamo(this, item.Item)` around line 278.